driver-core/src/main/java/com/datastax/driver/core/policies/DefaultRetryPolicy.java

```diff
@@ -135,4 +135,17 @@ public RetryDecision onUnavailable(Statement statement, ConsistencyLevel cl, int
             ? RetryDecision.tryNextHost(cl)
             : RetryDecision.rethrow();
     }
+
+    /**
+     * {@inheritDoc}
+     * <p>
+     * This implementation triggers a retry on the next host in the query plan,
+     * regardless of the consistency level, the number of retries or
+     * the possibility that the mutation has been applied server-side.
+     */
+    @Override
+    public RetryDecision onUnexpectedError(Statement statement, ConsistencyLevel cl, int nbRetry, boolean mightHaveBeenApplied) {
```

Comment:
Sorry if I just missed the conversation, but why not also provide the Exception?  Or is that considered too specific/too much information to make a decision here?